### PR TITLE
core[patch]: Document messages namespace

### PR DESCRIPTION
**Pull request by eyurtsev**
- Moved doc-strings below attribtues in TypedDicts -- seems to render better on APIReference pages.
* Provided more description and some simple code examples

### CODE REVIEW

File: libs/core/langchain_core/messages/ai.py [hunk 1]
```diff
@@ -25,19 +25,36 @@
...
+
+            {
+                "input_tokens": 10,
+                "output_tokens": 20,
+                "total_tokens": 30
+            }
     """
 
     input_tokens: int
+    """Count of input (or prompt) tokens."""
```
Comment (eyurtsev) [inline libs/core/langchain_core/messages/ai.py:42]: @baskaryan let me know if OK -- it seems to render better in APIReference

File: libs/core/langchain_core/messages/human.py [hunk 2]
```diff
@@ -4,11 +4,33 @@
...
+
+            # Instantiate a chat model and invoke it with the messages
+            model = ...
+            print(model.invoke(messages))
+    """
 
     example: bool = False
     """Whether this Message is being passed in to the model as part of an example 
-        conversation.
+    conversation.
```
Comment (eyurtsev) [inline libs/core/langchain_core/messages/human.py:33]: @baskaryan do you know if example is actually used / respected in any models?

File: libs/core/langchain_core/messages/system.py [hunk 3]
```diff
@@ -4,8 +4,29 @@
...
+
+    Example:
+
+        .. code-block:: python
+
+            from langchain_core.messages import HumanMessage, SystemMessage
+
+            messages = [
+                SystemMessage(
+                    content="You are a helpful assistant! Your name is Bob."
```
Comment (baskaryan) [inline libs/core/langchain_core/messages/system.py:20]: nit don't need to specify content as kwarg

Comment (baskaryan) [reply libs/core/langchain_core/messages/system.py:20]: (and imo cleaner not to)

File: libs/core/langchain_core/messages/human.py (hunk 2 above)
Comment (baskaryan) [reply libs/core/langchain_core/messages/human.py:33]: was for some of the original google palm models but afaict doesn't apply anymore

**Review by baskaryan — APPROVED** [review]

File: libs/core/langchain_core/messages/system.py (hunk 3 above)
Comment (eyurtsev) [reply libs/core/langchain_core/messages/system.py:20]: Most of our examples still do -- I'd prefer doing an upgrade a bit later once more folks are on the newest version of core?

**Pull request merged**